test_runner/regress/test_timeline_archive.py

```diff
@@ -114,3 +117,74 @@ def test_timeline_archive(neon_env_builder: NeonEnvBuilder, shard_count: int):
         leaf_timeline_id,
         state=TimelineArchivalState.UNARCHIVED,
     )
+
+
+@pytest.mark.parametrize("manual_offload", [False, True])
+def test_timeline_offloading(neon_env_builder: NeonEnvBuilder, manual_offload: bool):
```

Comment:
This only tests the happy path. Would it be useful to also test that offloading isn't done for unarchived branches and branches with active children?

Reply:
Added an assert for the absence of an offloading message for the main timeline.

For branches with active children, it's harder to create a testable situation: we already prevent archival of those branches in the first place. The second check is only for sanity.

Reply:
> For branches with active children, it's harder to create a testable situation: we already prevent archival of those branches in the first place. The second check is only for sanity.

Well, we do also only offload once all children have been offloaded. We do that because otherwise, in the timeline object of the child, the parent timeline would be removed from the tenant object and shut down, so dangling.

For that, I could probably request to offload the parent first, and see that there is an error.

Reply:
done